src/builder/command.rs

```diff
@@ -4241,7 +4241,9 @@ impl Command {
                 .action(ArgAction::Help);
             if self.long_help_exists {
                 arg = arg
-                    .help("Print help information (use --help for more detail)")
+                    // keep this message brief, for applications that want to
+                    // use the two-column short help format
+                    .help("Show help (--help for more)")
```

Comment:
> Applications may want to constrain their short help to use the two-column format so the -h output is as compact as possible. They might do this by optimizing short help texts to fit within the right-hand column on 80-character terminals, and then providing long help texts with more info. However, https://github.com/clap-rs/clap/pull/4159 made the short help text for -h/--help much longer when the short and long help are different, forcing short help to wrap to the multiple-line format if medium-sized option/value names are used.

Please open an issue for this.  I would like to discuss this further but find discussing requirements to be ineffective in PRs because we can end up with more than one PR that tries to address an issue and then the conversation gets scattered.